plugins/InspectionGadgets/src/com/siyeh/ig/redundancy/RedundantStringOperationInspection.java

```diff
@@ -159,6 +165,80 @@ private ProblemDescriptor getStringConstructorProblem(PsiNewExpression expressio
       return null;
     }
 
+    private ProblemDescriptor getRedundantCaseEqualsProblem(PsiMethodCallExpression call) {
+
+      PsiExpression equalTo = call.getArgumentList().getExpressions()[0];
+
+      //case: "foo".equals(s.toLowerCase())
+      if (equalTo instanceof PsiMethodCallExpression) {
+        PsiMethodCallExpression equalsToCallExpression = (PsiMethodCallExpression)equalTo;
+        if (isChangeCaseCall(equalsToCallExpression) &&
+            ExpressionUtils.computeConstantExpression(call.getMethodExpression().getQualifierExpression()) instanceof String) {
```

Comment:
I believe, `PsiUtil#isConstantExpression` would be faster (the fact that the qualifier is a String is already known by isChangeCaseCall)

Reply:
Thank you. I changed to PsiUtil#isConstantExpression